File: Inversion/serial/NA/src/hv_subs/PROGRAMS.330/CALPLOT/src/clib/shadep.c
/*
c---------------------------------------------------------------------c
c                                                                     c
c      COMPUTER PROGRAMS IN SEISMOLOGY                                c
c      VOLUME I                                                       c
c                                                                     c
c      PROGRAM: SHADEP                                                c
c                                                                     c
c      COPYRIGHT (C)  1997 R. B. Herrmann                             c
c                                                                     c
c      Department of Earth and Atmospheric Sciences                   c
c      Saint Louis University                                         c
c      3507 Laclede Avenue                                            c
c      St. Louis, Missouri 63103                                      c
c      U. S. A.                                                       c
c                                                                     c
c---------------------------------------------------------------------c
*/

#include <stdlib.h>
#include <stddef.h>
#include "callib.h"
#include "dosubs.h"
#include "calplot.h"
extern struct Gcplot Gcplt;


void shadep(int narr, float *xarr, float *yarr)
{
	float xx, yy;
	INT i;
	INT n;
	INT ilw, jlw;
	INT *x, *y;
	if((x = (INT *)calloc(narr, sizeof(INT))) == NULL)
		return;
	if((y = (INT *)calloc(narr, sizeof(INT))) == NULL)
		return;
	n = narr;
	for(i = 0 ; i < n ; i++){
       		xx = 1000. * (xarr[i]*Gcplt.xstp + Gcplt.xold);
       		yy = 1000. * (yarr[i]*Gcplt.ystp + Gcplt.yold);
		if(xx > 1000000000.0)
			ilw = 1000000000;
		else if(xx < -1000000000.0)
			ilw = -1000000000;
		else
			ilw = xx ;
		if(yy > 1000000000.0)
			jlw = 1000000000;
		else if(yy < -1000000000.0)
			jlw = -1000000000;
		else
			jlw = yy ;
		x[i] = ilw;
		y[i] = jlw;
	}
	(*do_fillp)(n,x,y);
	free(x);
	free(y);
}

File: Inversion/parallel/NA/src/hv_subs/PROGRAMS.330/CALPLOT/src/flib/shadep.f
c---------------------------------------------------------------------c
c                                                                     c
c      COMPUTER PROGRAMS IN SEISMOLOGY                                c
c      VOLUME I                                                       c
c                                                                     c
c      PROGRAM: SHADEP                                                c
c                                                                     c
c      COPYRIGHT (C)  1997 R. B. Herrmann                             c
c                                                                     c
c      Department of Earth and Atmospheric Sciences                   c
c      Saint Louis University                                         c
c      3507 Laclede Avenue                                            c
c      St. Louis, Missouri 63103                                      c
c      U. S. A.                                                       c
c                                                                     c
c---------------------------------------------------------------------c
        subroutine shadep(n, xarr, yarr)
c-----
c       shade a polygon
c-----
c       n       - number of x,y coordinate pairs
c       xarr    - x-array of abscissa
c       yarr    - y-array of ordinates
c-----
        common/Xcplot/xold,yold,xcur,ycur
        common/Ycplot/xstp,ystp
c-----
c      xold and yold are absolute plotter coordinates
c      xcur and ycur are coordinates of point relative
c            to current plotter position in current
c            software units
c-----
        integer*4 n
        real*4 xarr(n), yarr(n)
c-----
c       since we must convert to integers for calplot and
c       since we need to alloc() space for the integer arrays
c       we let the C routine do all the work, and 
c       pass all necessary information to the C routine
c-----
        call dffilp(n,xarr,yarr,xold,yold,xcur,ycur,xstp,ystp)
        return
        end
